include/drivers/bluetooth/hci_driver.h

```diff
@@ -185,6 +185,18 @@ struct bt_hci_driver {
 	 * @return 0 on success or negative error number on failure.
 	 */
 	int (*send)(struct net_buf *buf);
+
+	/**
+	 * @brief HCI vendor-specific setup
+	 *
+	 * Executes vendor-specific commands sequence to initialize
+	 * BT Controller before BT Host executes Reset sequence.
+	 *
+	 * @return 0 on success or negative error number on failure.
+	 */
+#if defined(CONFIG_BT_HCI_SETUP)
```

Comment:
Should this #if be above the Doxygen comment? I'm not sure how Doxygen will parse this.

Reply:
I just checked. The doc comment needs to be inside the #if. Otherwise the doc comment is applied to the next member (concatenated with that members doc comment) when the #if predicate is false.

Reply:
Also, this member does not appear in the Doxygen output because CONFIG_BT_HCI_SETUP is not defined during doc generation. How do we handle this?

Reply:
let me check how similar stuff is handled

Reply:
@npal-cy @alwa-nordic such situations are handled through the `PREDEFINED` list that's defined in `doc/zephyr.doxyfile.in`

Reply:
However the general preference is to avoid #ifdefs in header files to keep that list as small as possible. E.g. you definitely shouldn't do that for normal API functions that may or may not be available (in fact it prevents using IS_ENABLED() when calling them if the header file uses #ifdef). In this case however, since we decided to avoid the memory overhead when not needed, using the #ifdef in the header file is unavoidable.